src/codegen/llvm/llvm_benchmark.cpp

```diff
@@ -107,14 +107,15 @@ void LLVMBenchmark::run_benchmark(codegen::CodegenLLVMVisitor& visitor,
 
     // Benchmark every kernel.
     for (const auto& kernel_name: kernel_names) {
-        *log_stream << "Benchmarking kernel '" << kernel_name << "'\n";
+        // Initialise the data.
+        auto instance_data = codegen_data.create_data(instance_size, /*seed=*/1);
```

Comment:
Don't we want to allocate for every experiment to avoid caching for further iterations?

Reply:
Two aspects:
* In practice, we perform benchmark in two ways :   1) allocate small enough data so that it fits into particular cache level 2) allocate large enough data to avoid it fitting in particular cache level. For (2), we allocate dataset much greater than last level cache (LLC) so that we when we run multiple iterations we know that the data is streaming from DRAM.
* By the way - even if we allocate data on every iteration, the `initialisation of instance`  is  brining that into cache (if its small). So reallocation on each iteration doesn't really solve the caching aspect.